events-processor/cache/cache.go

```diff
@@ -0,0 +1,275 @@
+package cache
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/dgraph-io/badger/v4"
+	"github.com/getlago/lago/events-processor/config/database"
+	"github.com/getlago/lago/events-processor/utils"
+)
+
+// Cache wraps BadgerDB to provide an in-memory key-value store with JSON serialization
+// It manages the lifecycle of cached data and coordinates snapshot loading and CDC consumption.
+type Cache struct {
+	ctx    context.Context
+	db     *badger.DB
+	logger *slog.Logger
+	wg     sync.WaitGroup
+}
+
+// CacheConfig holds the configuration needed to initialize a new Cache instance.
+type CacheConfig struct {
+	Context context.Context
+	Logger  *slog.Logger
+}
+
+// NewCache creates and initializes a new in-memory cache instance.
+// It configures the database with default options
+func NewCache(config CacheConfig) (*Cache, error) {
+	opts := badger.DefaultOptions("").WithInMemory(true)
+	opts.Logger = nil
+
+	logger := config.Logger.With("pkg", "cache")
+
+	db, err := badger.Open(opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open badger db: %w", err)
+	}
+
+	return &Cache{
+		db:     db,
+		logger: logger,
+		ctx:    config.Context,
+	}, nil
+}
+
+func (c *Cache) Close() error {
+	return c.db.Close()
+}
+
+func (c *Cache) Wait() {
+	c.wg.Wait()
+}
+
+func (c *Cache) LoadInitialSnapshot() {
+	dbConfig := database.DBConfig{
+		Url:      os.Getenv("DATABASE_URL"),
+		MaxConns: 10,
+	}
+
+	db, err := database.NewConnection(dbConfig)
+	if err != nil {
+		c.logger.Error("Error connecting to the database", slog.String("error", err.Error()))
+		utils.CaptureError(err)
+		panic(err.Error())
```

Comment:
You can now do this:
```suggestion
        utils.LogAndPanic(lc.ogger, err, "Error connecting to the database")
```